scorescreen.py

```diff
@@ -21,98 +21,112 @@
 # Utkarsh Tiwari    iamutkarshtiwari@gmail.com
 
 import os
-import gi
-gi.require_version('Gtk', '3.0')
-from gi.repository import Gtk
 import pickle
-import pygame
 import sys
 from gettext import gettext as _
 
+import pygame
+import gi
+gi.require_version('Gtk', '3.0')
+from gi.repository import Gtk
 
-class scorescreen:
+from sugar3.activity.activity import get_activity_root
 
-    def run(self, g, scores):
-        black = (0, 0, 0)
-        white = (255, 255, 255)
-        red = (255, 0, 0)
-        clock = pygame.time.Clock()
-        crashed = False
-        press = 0
-        land1x = 350
-        scorescreen = pygame.image.load("assets/scorescreen.png").convert()
-        scorescreen = pygame.transform.scale(scorescreen, (490, 768))
-        restart = pygame.image.load("assets/restart.png")
-        restart = pygame.transform.scale(restart, (120, 120))
-
-        # font load
-        font2 = pygame.font.Font("fonts/Arimo.ttf", 30)
-        font3 = pygame.font.Font("fonts/Arimo.ttf", 80)
-
-        # Scores load
-	if os.path.exists("score.pkl")==False:
-	    open('score.pkl','w+')
-
-        if os.path.getsize("score.pkl") == 0:
-            with open('score.pkl', 'wb') as output:
+FPS = 60
+BLACK = (0, 0, 0)
+WHITE = (255, 255, 255)
+
+SCORESCREEN_PATH = os.path.join("assets", "scorescreen.png")
+RESTART_IMAGE_PATH = os.path.join("assets", "restart.png")
+HOME_IMAGE_PATH = os.path.join("assets", "home.png")
+FONT_PATH = os.path.join("fonts", "Arimo.ttf")
+
+
+class Scorescreen:
+    def __init__(self, game):
+        self.running = True
+
+        self.scorescreen = self.load_scaled_image(
+            SCORESCREEN_PATH, (490, 768))
+        self.restart = self.load_scaled_image(
+            RESTART_IMAGE_PATH, (120, 120))
+        self.load_fonts(game)
+
+        self.score_path = os.path.join(
+            get_activity_root(), 'data', 'score.pkl')
+
+    def load_fonts(self, game):
+        self.font2 = pygame.font.Font(
+            FONT_PATH, 30)
+        self.font3 = pygame.font.Font(
+            FONT_PATH, 80)
+
+    # Load and Scale image
+    def load_scaled_image(self, path, size):
+        image = pygame.image.load(path)
+        width, height = size
+        scaled_image = pygame.transform.scale(
+            image.convert_alpha(), (int(width), int(height)))
+        return scaled_image
+
+    def read_max_score(self):
+        if not os.path.exists(self.score_path):
+            open(self.score_path, 'w+')
```

Comment:
Why do this when self.score_path isn't assigned to a variable here but just opened and won't be closed as it doesn't have a reference, plus you use a context manager below to access the file and that handles creation if the file doesn't exist.

Reply:
Fixed